source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -194,7 +202,7 @@ void DnsResolverImpl::AddrInfoPendingResolution::onAresGetAddrInfoCallback(
     //
     // The channel cannot be destroyed and reinitialized here because that leads to a c-ares
     // segfault.
-    if (status == ARES_ECONNREFUSED) {
+    if (status == ARES_ECONNREFUSED || status == ARES_EREFUSED) {
```

Comment:
This change is for updating c-ares library to 1.20, but have nothing to do with the udp_max_queries option support.  Does it make sense to separate it?

Reply:
`DestroyChannelOnRefused` and `CustomResolverValidAfterChannelDestruction` test fail in `dns_impl_test.cc` after upgrading the c-ares library to 1.20 if this isn't added, while `udp_max_queries` is supported in 1.20.

Reply:
Can you split into two PRs? It will make it easier to review the dependency updates vs. the UDP query specific change proposed. Thanks!

Reply:
sure I can split it. Let me send out a PR for library update.